02_bubble_sort:selva/bubblesort.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(items []int) []int {
+
+	if len(items) > 1 {
+		pivotIndex := len(items) / 2
+		var smallerItems = []int{}
+		var largerItems = []int{}
+
+		for i := range items {
+			val := items[i]
+			if i != pivotIndex {
+				if val < items[pivotIndex] {
+					smallerItems = append(smallerItems, val)
+				} else {
+					largerItems = append(largerItems, val)
+				}
+			}
+		}
+
+		bubble(smallerItems)
+		bubble(largerItems)
+
+		var merged = append(append(append([]int{}, smallerItems...), []int{items[pivotIndex]}...), largerItems...)
```

Comment:
`[]int{items[pivotIndex]}...` is the same as `items[pivotIndex]`. But it's not required anyway if you allow the pivot element to drop into `largerItems`, as recommended above.